include/keepkey/firmware/cosmos.h

```diff
@@ -0,0 +1,51 @@
+#ifndef __COSMOS_H__
+#define __COSMOS_H__
+
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+#include "crypto.h"
+#include "fsm.h"
+#include "messages.pb.h"
+#include "trezor/crypto/bip32.h"
+
+bool cosmos_path_mismatched(const CoinType *_coin,
+                            const uint32_t *_address_n,
+                            const uint32_t _address_n_count);
+bool cosmos_getAddress(const HDNode *_node, char *address);
+bool cosmos_signTxInit(const HDNode* _node,
+                       const uint32_t _address_n[8],
```

Comment:
The `8` isn't enforced unless you do `[static 8]`. Better to make it a `const uint32_t *_address_n`, and let `address_n_count` do its thing.